src/Shared/HttpSys/RequestProcessing/RequestUriBuilder.cs

```diff
@@ -28,6 +28,13 @@ public static string DecodeAndUnescapePath(Span<byte> rawUrlBytes)
                 return "/";
             }
 
+            // OPTIONS *
+            // RemoveDotSegments Asserts path always starts with a '/'
+            if (rawPath.Length == 1 && rawPath[0] == (byte)'*')
```

Comment:
Random?

Reply:
Some recent changes caused a Debug.Assert that prevents the tests from running locally. The CI doesn't check Asserts.